x/twap/logic_test.go

```diff
@@ -557,330 +547,14 @@ type computeThreeAssetArithmeticTwapTestCase struct {
-				osmoassert.DecApproxEq(t, tc.expTwap, actualTwap, osmomath.GetPowPrecision())
-			})
-		})
-	}
-}
-
-func TestComputeArithmeticTwap_ThreeAsset_Arithmetic(t *testing.T) {
-	tenSecAccum := OneSec.MulInt64(10)
-	pointOneAccum := OneSec.QuoInt64(10)
-	tests := map[string]computeThreeAssetArithmeticTwapTestCase{
-		"three asset basic: spot price = 1 for one second, 0 init accumulator": {
-			startRecord: newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			endRecord:   newThreeAssetOneSidedRecord(tPlusOne, OneSec, true),
-			quoteAsset:  []string{denom0, denom0, denom1},
-			expTwap:     []sdk.Dec{sdk.OneDec(), sdk.OneDec(), sdk.OneDec()},
-		},
-		"three asset same record: asset1, end spot price = 1": {
-			startRecord: newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			endRecord:   newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			quoteAsset:  []string{denom1, denom2, denom2},
-			expTwap:     []sdk.Dec{sdk.OneDec(), sdk.OneDec(), sdk.OneDec()},
-		},
-		"three asset. accumulator = 10*OneSec, t=5s. 0 base accum": testThreeAssetCaseFromDeltas(
-			sdk.ZeroDec(), tenSecAccum, 5*time.Second, sdk.NewDec(2)),
-
-		// test that base accum has no impact
-		"three asset. accumulator = 10*OneSec, t=5s. 10 base accum": testThreeAssetCaseFromDeltas(
-			sdk.NewDec(10), tenSecAccum, 5*time.Second, sdk.NewDec(2)),
-		"three asset. accumulator = 10*OneSec, t=100s. .1*second base accum": testThreeAssetCaseFromDeltas(
-			pointOneAccum, tenSecAccum, 100*time.Second, sdk.NewDecWithPrec(1, 1)),
-	}
-	for name, test := range tests {
-		t.Run(name, func(t *testing.T) {
-			for i, startRec := range test.startRecord {
-				actualTwap, err := twap.ComputeTwap(startRec, test.endRecord[i], test.quoteAsset[i], twap.ArithmeticTwapType)
-				require.Equal(t, test.expTwap[i], actualTwap)
-				require.NoError(t, err)
-			}
-		})
-	}
-}
-
-func TestComputeArithmeticTwap_ThreeAsset_Geometric(t *testing.T) {
-	var (
-		five        = sdk.NewDec(5)
-		fiveFor3Sec = OneSec.MulInt64(3).Mul(twap.TwapLog(five))
-
-		ten          = five.MulInt64(2)
-		tenFor100Sec = OneSec.MulInt64(100).Mul(twap.TwapLog(ten))
-
-		errTolerance = sdk.MustNewDecFromStr("0.00000001")
-	)
-
-	tests := map[string]computeThreeAssetArithmeticTwapTestCase{
-		"three asset basic: spot price = 10 for one second, 0 init accumulator": {
-			startRecord: newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			endRecord:   newThreeAssetOneSidedRecord(tPlusOne, geometricTenSecAccum, true),
-			quoteAsset:  []string{denom0, denom0, denom1},
-			expTwap:     []sdk.Dec{sdk.NewDec(10), sdk.NewDec(10), sdk.NewDec(10)},
-		},
-		"three asset same record: asset1, end spot price = 1": {
-			startRecord: newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			endRecord:   newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			quoteAsset:  []string{denom1, denom2, denom2},
-			expTwap:     []sdk.Dec{sdk.OneDec(), sdk.OneDec(), sdk.OneDec()},
-		},
```

Comment:
Note: confirmed that all test cases but this one have been moved to: TestComputeArithmeticTwap_ThreeAsset_Geometric (pending rename)

This test case is now obsolete. We test similar behavior here: https://github.com/osmosis-labs/osmosis/pull/3696/files#diff-022ef0e75aa0348c4eff672e6fba120ba0e51a091218ca5651efbefe536cecbdR33